Expanding on sentence case to everywhere else

karmatosed opened the pull request:
So far we have only done this in the tooltips and in menus. If sentance case of right approach we should consider taking this across all areas.

Included in this:

- Side panel
- Block library
- Settings

There may be areas missed but gives a starting view to see if everyone agrees to go forward with this across everything. If we do, then we need a patch for outside the editor.

Expands on #16764

Commit:
Expanding on sentance case to everywhere else

So far we have only done this in the tooltips and in menus. If sentance case of right approach we should consider taking this across all areas.

Included in this:

- Side panel
- Block library
- Settings

There may be areas missed, but gives a starting view to see if everyone agrees to go forward with this across everything. If we do, then we need a patch for core.

Expands on #16764

karmatosed commented:
Some images with these changes:

![image](https://user-images.githubusercontent.com/253067/69651539-287ac880-1068-11ea-9e76-f104452422eb.png)

![image](https://user-images.githubusercontent.com/253067/69651563-30d30380-1068-11ea-86c2-ac7f392c7203.png)

Commit:
Catch a few missed

Commit:
Update tag

enriquesanchez commented:
I agree this is much better and happy to see the work that started on #16764 expand to other areas.

karmatosed commented:
@noisysocks I don't know if you can help here but the tests seem stuck again :(

mkaz commented:
@karmatosed This time they are valid tests fails. The renaming of the default categories "Common blocks", "Layout elements", and "Reusable blocks" is causing some inserter menu test to fail.

Tests at: packages/block-editor/src/components/inserter/test/menu.js 

Likewise with the description in this test: 
FAIL packages/edit-post/src/components/keyboard-shortcut-help-modal/test/index.js

And label in this test:
FAIL packages/edit-post/src/components/options-modal/test/index.js 

You can see the error messages here: https://travis-ci.com/WordPress/gutenberg/jobs/260658429

Commit:
Test catches

Fixes test issues.

Commit:
Catches the test fails for sentance case.

Commit:
Test fix

Commit:
Update index.js

Commit:
Catch some more test fails

Commit:
Test fixes

mcsf committed:
Override WP labels for post and page types

… to conform to sentence-style capitalisation across the editor UI.

(This is a quick commit to show the idea. Do expand on it and polish.)

From Slack: "If we used this it would have to be properly marked [e.g.
@TODO] and commented so that before the next core release we could
revisit these labels as a whole."

youknowriad commented on lib/client-assets.php:
```diff
@@ -149,6 +149,14 @@ function gutenberg_override_translation_file( $file, $handle ) {
 add_filter( 'load_script_translation_file', 'gutenberg_override_translation_file', 10, 2 );
 
+function gutenberg_override_posttype_labels( $labels ) {
+	$labels->featured_image = 'Feaaatured image';
```
looks like the macbook keyboard is not working properly :)

mcsf replied:
Oops :D

Soean commented on lib/client-assets.php:
```diff
@@ -149,6 +149,25 @@ function gutenberg_override_translation_file( $file, $handle ) {
+ * @return object Object with all the labels, including overridden ones.
+ */
+function gutenberg_override_posttype_labels( $labels ) {
+	$labels->featured_image = 'Featured image';
```
The string needs to be translatable.

Soean commented on lib/client-assets.php:
```diff
@@ -149,6 +149,25 @@ function gutenberg_override_translation_file( $file, $handle ) {
+ *
+ * @return object Object with all the labels, including overridden ones.
+ */
+function gutenberg_override_posttype_labels( $labels ) {
```
Why do we need this function in Gutenberg?

karmatosed replied:
@Soean the problem is that our labels come from core so we can't have sentence case without. I will loop @mcsf in here as they worked on this solution to help.

Soean replied:
Then we have to change it in Core :-)

mcsf replied:
@Soean, thanks for that watchful eye. :) Indeed, we need to change it in core.

c7f1e892 was a drive-by commit to unblock Tammie. The commit message itself did mention the need to address this in core, but commit messages aren't very visible here. This all came from [this thread in Slack](https://wordpress.slack.com/archives/C02QB2JS7/p1576067206315600), which also acknowledged the need to address the root cause.

The question that this commit raised is: in the interest of landing something consistent here in #18758, _can we_ add this filter for the plugin, provided that we also open the appropriate Trac tickets?

Soean replied:
Thanks for clarification. 
If a post type has a different label than "Featured image" for example "Book cover", then the filter overrides the value? This could be a problem.

mcsf replied:
>  If a post type has a different label than "Featured image" for example "Book cover", then the filter overrides the value? This could be a problem.

This change is only targeting `post` and `page` post types. Do you think that, even then, we should guard against potential customisations? Something like:

```php
if ( __( 'Featured Image' ) == $labels->featured_image ) {
  $labels->featured_image = __( 'Featured image', 'gutenberg' );
}
return $labels;
```

karmatosed replied:
@Soean can you confirm if what @mcsf has done is ok for now? I will happily make a trac ticket, but it would be good to get a start on things in Gutenberg here.

Soean replied:
Yes, now it's translateable and it works for post und page post type. 
So we can merge it and link it in the trac ticket, so we can remove it after merging it in core.

jorgefilipecosta replied:
Hi @mcsf, it seems this change needs to be backported to core. Would you be able to submit a core patch?

mcsf replied:
Will do, @jorgefilipecosta.

mcsf replied:
→ https://core.trac.wordpress.org/ticket/49371

mcsf replied:
Done

mcsf committed:
gutenberg_override_posttype_labels: small fixes

mcsf committed:
Merge branch 'master' into try/sentance-everywhere

Commit:
Merge branch 'master' into try/sentance-everywhere
